_posts/2022-12-16-aave-rebasing.md

```diff
@@ -0,0 +1,332 @@
+
+Initially, this cancels out the effect of scaling down the balance, so that if
+someone went to withdraw immediately after depositing they would get all of
+their money back. E.g. a stored balance of
+98.93e18 aPolyDAI would be multiplied by the liquidity index of 1.01083, resulting a
+withdrawable balance of 100 DAI.
+
+But over time, as the liquidity index increases, it results in a net positive aToken balance, since the stored balance is being multiplied by a larger number than the deposit was initially divided by.
+
+#### A Toy Example
+
+Let's put these concepts together and see how interest works in a bigger
+context.
+
+* suppose some asset has earned 5% interest over its lifetime in Aave
+* the liquidity index for this asset will be 5%, or 1.05e27
+* Ben deposits $100 of the asset
+* the aToken scales down his deposit by the liquidity index and stores the result,
+  in this case 100 / 1.05 == 95.23
+* if Ben immediately withdrew his money, his rebased balance would be: `storedBalance * liquidityIndex`, i.e. 95.23 * 1.05 == $100, what we expect
+* a year goes by and the pool earns another 5% interest, so the new liquidity index is 1.05 * 1.05 == 1.1025
+* I now deposit $100 of the asset
+* once again, the aToken scales down my deposit and stores the result: 100 / 1.1025 == 90.70 is my balance in storage
+* if I immediately withdrew, my rebased balance would be 90.70 * 1.1025 == $100, i.e. exactly what I put in
+* Ben's rebased balance at this point would be 95.23 * 1.1025 == $105
+* so he'd have 5% more underlying than me, as expected (since the pool as a whole earned 5% while he had his money in it)
+
+And that's it! That is how interest works in Aave.
+
+### Flexible Voting
+
+[Flexible voting](https://www.scopelift.co/blog/introducing-flexible-voting) (or _flex voting_ for short)
+is a novel extension to Compound-style governance systems.
+It allows delegates to split their voting weight across For/Against/Abstain options, rather than having to put all of it behind one.
+
+If flex voting could be added to aTokens, this would unlock a large amount of
+value for Aave.
+
+To see why, imagine you hold UNI.
+UNI gets its intrinsic value from being the voting token for [Uniswap governance proposals](https://app.uniswap.org/#/vote).
+Anyone who holds UNI can delegate and vote on these proposals and help set the course of one of the biggest projects in crypto.
+
+So the main value of UNI comes from its use in voting.
+But it would be really nice if you could also deposit your UNI to earn yield at the same time.
+
+The trouble is that if you were to deposit UNI into a DeFi protocol (like Aave) you would lose your voting power.
+This is because the address that holds the tokens controls their voting weight.
+And if you deposit your UNI, the receiving contract -- not you -- would hold the tokens.
+
+So governance token holders are forced to choose: either participate in governance or earn yield.
+They cannot currently have both.
```

Comment:
They can if they deposit into element finance and use a voting vault. I don't know much about those, but maybe soften this sentence a bit as a result?
```suggestion
They cannot currently have both.
```

Reply:
I softened this language a bit in 8468901